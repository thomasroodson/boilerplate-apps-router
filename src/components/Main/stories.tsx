import { Meta, StoryObj } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  argTypes: {
    title: {
      control: { type: 'text' }
    },
    description: {
      control: { type: 'text' }
    }
  },
  parameters: {
    docs: {
      description: {
        component: 'Another description, overriding the comments'
      }
    }
  }
} as Meta

export const Default: StoryObj = {}

export const Basic: StoryObj = {
  args: {
    title: 'Title Basic',
    description: 'Description Basic'
  }
}
