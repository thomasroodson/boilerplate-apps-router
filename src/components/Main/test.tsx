import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // rederizar o component
    const { container } = render(<Main />)

    // busca o elemento e verifica a existência dele
    expect(
      screen.getByRole('heading', { name: /react avançado/i })
    ).toBeInTheDocument()

    //gerar um snapshot
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the color correctly', () => {
    // rederizar o component
    const { container } = render(<Main />)

    // verifica se o background-color está correto
    expect(container.firstChild).toHaveStyle({ 'background-color': '#028090' })
  })
})
