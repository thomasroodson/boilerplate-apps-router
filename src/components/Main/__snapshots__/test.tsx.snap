// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #028090;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-top: 2rem;
  margin-bottom: 3rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(60rem, 100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um computador"
    class="c4"
    src="/img/ilustracao.svg"
  />
</main>
`;
